Sources/ClientRuntime/Message/URI.swift

```diff
@@ -0,0 +1,273 @@
+        queryItems += items.toURLQueryItems()
+
+        if queryItems.containsPercentEncode() {
+            self.urlComponents.percentEncodedQueryItems = queryItems
+        } else {
+            self.urlComponents.queryItems = queryItems
+        }
+
+        return self
+    }
+
+    @discardableResult
+    public func appendQueryItem(_ item: SDKURLQueryItem) -> URIBuilder {
+        self.appendQueryItems([item])
+        return self
+    }
+
+    @discardableResult
+    public func withUsername(_ value: String?) -> URIBuilder {
+        if let username = value {
+            if username.isPercentEncoded {
+                self.urlComponents.percentEncodedUser = username
+            } else {
+                self.urlComponents.user = username
+            }
+        }
+        return self
+    }
+
+    @discardableResult
+    public func withPassword(_ value: String?) -> URIBuilder {
+        if let password = value {
+            if password.isPercentEncoded {
+                self.urlComponents.percentEncodedPassword = password
+            } else {
+                self.urlComponents.password = password
+            }
+        }
+        return self
+    }
+
+    @discardableResult
+    public func withFragment(_ value: String?) -> URIBuilder {
+        if let fragment = value {
+            if fragment.isPercentEncoded {
+                self.urlComponents.percentEncodedFragment = fragment
+            } else {
+                self.urlComponents.fragment = fragment
+            }
+        }
+        return self
+    }
+
+    public func build() -> URI {
+        return URI(scheme: Scheme(rawValue: self.urlComponents.scheme!)!,
+                   path: self.urlComponents.percentEncodedPath,
+                   host: self.urlComponents.percentEncodedHost!,
+                   port: self.urlComponents.port.map { Int16($0) },
+                   queryItems: self.urlComponents.percentEncodedQueryItems?.map {
+                        SDKURLQueryItem(name: $0.name, value: $0.value)
+                   } ?? [],
+                   username: self.urlComponents.percentEncodedUser,
+                   password: self.urlComponents.percentEncodedPassword,
+                   fragment: self.urlComponents.percentEncodedFragment)
+    }
+
+    // We still have to keep 'url' as an optional, since we're
+    // dealing with dynamic components that could be invalid.
+    fileprivate func getUrl() -> URL? {
+        let isInvalidHost = self.urlComponents.host?.isEmpty ?? false
+        return isInvalidHost && self.urlComponents.path.isEmpty ? nil : self.urlComponents.url
+    }
+}
+
+extension String {
+    var isPercentEncoded: Bool {
+        let decoded = self.removingPercentEncoding
+        return decoded != nil && decoded != self
+    }
+
+    public func percentEncodePathIncludingSemicolon() -> String {
+        let allowed =
+            // swiftlint:disable:next force_cast
```

Comment:
The disabled lint category needs to be enabled after the relevant changes. Otherwise the lint category stays disabled for the entire file. I see the force cast only happens once in this file, but still should have the enable statement afterwards to future-proof it.